src/Sulu/Component/Content/Document/Syncronization/DocumentRegistrator.php

```diff
@@ -0,0 +1,214 @@
+            }
+
+            $this->registerSingleDocumentWithPDM($propertyValue);
+        }
+
+        // TODO: Workaround for the fact that "parent" is not in the metadata,
+        // see: https://github.com/sulu-io/sulu-document-manager/issues/67
+        if ($document instanceof ParentBehavior) {
+            if ($parent = $document->getParent()) {
+                $this->registerSingleDocumentWithPDM($parent, true);
+            }
+        }
+    }
+
+    private function registerSingleDocumentWithPDM($document, $create = false)
+    {
+        $ddmInspector = $this->defaultManager->getInspector();
+        $pdmRegistry = $this->publishManager->getRegistry();
+
+        // if the PDM registry already has the document, then
+        // there is nothing to do - the document manager will
+        // handle the rest.
+        if (true === $pdmRegistry->hasDocument($document)) {
+            return;
+        }
+
+        // see if we can resolve the corresponding node in the PDM.
+        // if we cannot then we either return and let the document
+        // manager create the new node, or, if $create is true, create
+        // the missing node (this happens when registering a document
+        // which is a relation to the incoming DDM document).
+        if (false === $uuid = $this->resolvePDMUUID($document)) {
+            if (false === $create) {
+                return;
+            }
+
+            $this->publishManager->getNodeManager()->createPath(
+                $ddmInspector->getPath($document), $ddmInspector->getUuid($document)
+            );
+
+            return;
+        }
+
+        // register the DDM document against the PDM PHPCR node.
+        $node = $this->publishManager->getNodeManager()->find($uuid);
+        $locale = $ddmInspector->getLocale($document);
+        $pdmRegistry->registerDocument(
+            $document,
+            $node,
+            $locale
+        );
+    }
+
+    /**
+     * If possible, resolve the UUID of the node in the PDM corresponding to
+     * the DDM node.
+     *
+     * If the UUID does not exist, we check to see if the path exsits.
+     * if neither the path or UUID exist, then the PDM should create a new
+     * document and we ensure that the PARENT path exists and if it doesn't
+     * we syncronize the ancestor nodes from the DDM.
+     *
+     * We return FALSE in the case that a new document should be created.
+     *
+     * In the case the UUID does not exist we assume that in a valid system
+     * that path will also NOT EXIST. If the path does exist, then it means that
+     * the corresponding PHPCR nodes were created independently of each
+     * other and bypassed the syncrhonization system and we throw an exception.
+     *
+     * @throws \RuntimeException If the UUID could not be resolved and it would be
+     *                           invalid to implicitly allow the node to be created.
+     */
+    private function resolvePDMUUID($object)
```

Comment:
Still a lot more than just resolve something. E.g. the `NodeHelper::createPath` method does also not create a path, if it already exists.